pkg/reconciler/ksvc/resources/knative_configuration.go

```diff
@@ -24,59 +27,68 @@ func CreateModelServingContainer(modelName string, modelSpec *v1alpha1.ModelSpec
 }
 
 func CreateKnativeConfiguration(kfsvc *v1alpha1.KFService) (*knservingv1alpha1.Configuration, *knservingv1alpha1.Configuration) {
-	var canaryContainer *v1.Container
-	defaultContainer := CreateModelServingContainer(kfsvc.Name, &kfsvc.Spec.Default)
-	if kfsvc.Spec.Canary != nil {
-		canaryContainer = CreateModelServingContainer(kfsvc.Name, &kfsvc.Spec.Canary.ModelSpec)
-	}
 	annotations := make(map[string]string)
-
 	if kfsvc.Spec.MinReplicas != 0 {
 		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(kfsvc.Spec.MinReplicas)
 	}
 	if kfsvc.Spec.MaxReplicas != 0 {
 		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(kfsvc.Spec.MaxReplicas)
 	}
-	for k, v := range kfsvc.Annotations {
-		annotations[k] = v
-	}
+
 	defaultConfiguration := &knservingv1alpha1.Configuration{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      constants.DefaultConfigurationName(kfsvc.Name),
-			Namespace: kfsvc.Namespace,
-			Labels:    kfsvc.Labels,
+			Name:        constants.DefaultConfigurationName(kfsvc.Name),
+			Namespace:   kfsvc.Namespace,
+			Labels:      kfsvc.Labels,
+			Annotations: union(kfsvc.Annotations, annotations),
 		},
 		Spec: knservingv1alpha1.ConfigurationSpec{
 			RevisionTemplate: &knservingv1alpha1.RevisionTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: union(kfsvc.Labels, map[string]string{
+						constants.KFServicePodLabelKey: kfsvc.Name,
+					}),
+					Annotations: kfsvc.Annotations,
+				},
 				Spec: knservingv1alpha1.RevisionSpec{
-					Container: defaultContainer,
+					Container: CreateModelServingContainer(kfsvc.Name, &kfsvc.Spec.Default),
```

Comment:
any reason you inlined this? i think the previous style (of assigning to a variable) makes for  easier debugging

Reply:
I'm a huge fan of functional programming and in general keeping tightly scoped variables. I love that Golang's inline object instantiation makes this so easy. This pattern closely mirrors Kubernetes' declarative paradigm (well, as close as one could get), helps enforce immutability, and generally reduces code complexity. 

Of course, there's always a time and place for everything and I'm willing to be persuaded -- I certainly wouldn't trade data-plane performance for this type of pattern.

Reply:
My original intention is to leave Annotation as nil if nothing is set, when doing `DeepEqual` nil and empty  map shows as a diff, so when you get from cache it is nil but  here it sets to empty map. But I guess we can do semantic equals to avoid the issue.